giskard/models/base/model.py

```diff
@@ -593,3 +601,154 @@ def __str__(self) -> str:
         if self.name:  # handle both None and empty string
             return f"{self.name}({self.id})"
         return super().__str__()  # default to `<giskard.models.base.Model object at ...>`
+
+    def _get_available_tools(self, dataset: Dataset, scan_report: ScanReport) -> dict:
+        """Get the dictionary with available tools.
+
+        Parameters
+        ----------
+        dataset : Dataset
+            Giskard Dataset to be analysed by Tools.
+        scan_report : ScanReport
+            Giskard Scan Report to be analysed by Tools.
+
+        Returns
+        -------
+        dict[str, BaseTool]
+            The dictionary with Tools' names and related instances.
+        """
+        try:
+            from ...llm.talk.tools import (
+                IssuesScannerTool,
+                MetricTool,
+                PredictTool,
+                SHAPExplanationTool,
+            )
+        except ImportError as err:
+            raise GiskardInstallationError(flavor="talk") from err
+
+        return {
+            PredictTool.default_name: PredictTool(model=self, dataset=dataset),
+            MetricTool.default_name: MetricTool(model=self, dataset=dataset),
+            SHAPExplanationTool.default_name: SHAPExplanationTool(model=self, dataset=dataset),
+            IssuesScannerTool.default_name: IssuesScannerTool(scan_report=scan_report),
+        }
+
+    @staticmethod
+    def _gather_context(message_list: list) -> str:
```

Comment:
Could be nice to check that the resulting string is not too long for the context windows of the LLM